src/i16x16_.rs

```diff
@@ -0,0 +1,503 @@
+          self.arr[11].bitxor(rhs.arr[11]),
+          self.arr[12].bitxor(rhs.arr[12]),
+          self.arr[13].bitxor(rhs.arr[13]),
+          self.arr[14].bitxor(rhs.arr[14]),
+          self.arr[15].bitxor(rhs.arr[15]),
+        ]}
+      }
+    }
+  }
+}
+
+macro_rules! impl_shl_t_for_i16x16 {
+  ($($shift_type:ty),+ $(,)?) => {
+    $(impl Shl<$shift_type> for i16x16 {
+      type Output = Self;
+      /// Shifts all lanes by the value given.
+      #[inline]
+      #[must_use]
+      fn shl(self, rhs: $shift_type) -> Self::Output {
+        let u: u64 = rhs as u64;
+        pick! {
+          if #[cfg(target_feature="avx2")] {
+            let shift = cast([u, 0]);
+            Self { avx2: shl_all_u16_m256i(self.avx2, shift) }
+          } else if #[cfg(target_feature="sse2")] {
+            let shift = cast([u, 0]);
+            Self {
+              sse0: shl_all_u16_m128i(self.sse0, shift),
+              sse1: shl_all_u16_m128i(self.sse1, shift)
+            }
+          } else {
+            Self { arr: [
+              self.arr[0] << u,
+              self.arr[1] << u,
+              self.arr[2] << u,
+              self.arr[3] << u,
+              self.arr[4] << u,
+              self.arr[5] << u,
+              self.arr[6] << u,
+              self.arr[7] << u,
+              self.arr[8] << u,
+              self.arr[9] << u,
+              self.arr[10] << u,
+              self.arr[11] << u,
+              self.arr[12] << u,
+              self.arr[13] << u,
+              self.arr[14] << u,
+              self.arr[15] << u,
+            ]}
+          }
+        }
+      }
+    })+
+  };
+}
+impl_shl_t_for_i16x16!(i8, u8, i16, u16, i32, u32, i64, u64, i128, u128);
+
+macro_rules! impl_shr_t_for_i16x16 {
+  ($($shift_type:ty),+ $(,)?) => {
+    $(impl Shr<$shift_type> for i16x16 {
+      type Output = Self;
+      /// Shifts all lanes by the value given.
+      #[inline]
+      #[must_use]
+      fn shr(self, rhs: $shift_type) -> Self::Output {
+        let u: u64 = rhs as u64;
+        pick! {
+          if #[cfg(target_feature="avx2")] {
+            let shift = cast([u, 0]);
+            Self { avx2: shr_all_u16_m256i(self.avx2, shift) }
+          } else if #[cfg(target_feature="sse2")] {
+            let shift = cast([u, 0]);
+            Self { sse0: shr_all_i16_m128i(self.sse0, shift),
+              sse1: shr_all_i16_m128i(self.sse1, shift) }
+          } else {
+            Self { arr: [
+              self.arr[0] >> u,
+              self.arr[1] >> u,
+              self.arr[2] >> u,
+              self.arr[3] >> u,
+              self.arr[4] >> u,
+              self.arr[5] >> u,
+              self.arr[6] >> u,
+              self.arr[7] >> u,
+              self.arr[8] << u,
+              self.arr[9] << u,
+              self.arr[10] << u,
+              self.arr[11] << u,
+              self.arr[12] << u,
+              self.arr[13] << u,
+              self.arr[14] << u,
+              self.arr[15] << u,
```

Comment:
8 to 15 are still shifting left

Reply:
ok will add more tests to pick that up.

Reply:
(good spot)

Reply:
Ah I see that the test was there and it was caught by CI. nice.